homeassistant/components/recorder/statistics.py

```diff
@@ -481,66 +465,37 @@ def compile_hourly_statistics(
             }
 
     # Get last hour's last sum
-    if instance._db_supports_row_number:  # pylint: disable=[protected-access]
-        subquery = (
-            session.query(*QUERY_STATISTICS_SUMMARY_SUM)
-            .filter(StatisticsShortTerm.start >= bindparam("start_time"))
-            .filter(StatisticsShortTerm.start < bindparam("end_time"))
-            .subquery()
-        )
-        query = (
-            session.query(subquery)
-            .filter(subquery.c.rownum == 1)
-            .order_by(subquery.c.metadata_id)
-        )
-        stats = execute(query.params(start_time=start_time, end_time=end_time))
-
-        if stats:
-            for stat in stats:
-                metadata_id, start, last_reset, state, _sum, _ = stat
-                if metadata_id in summary:
-                    summary[metadata_id].update(
-                        {
-                            "last_reset": process_timestamp(last_reset),
-                            "state": state,
-                            "sum": _sum,
-                        }
-                    )
-                else:
-                    summary[metadata_id] = {
-                        "start": start_time,
-                        "last_reset": process_timestamp(last_reset),
-                        "state": state,
-                        "sum": _sum,
-                    }
-    else:
-        stmt = _compile_hourly_statistics_summary_sum_legacy_stmt(start_time, end_time)
-        stats = execute_stmt_lambda_element(session, stmt)
+    subquery = (
+        session.query(*QUERY_STATISTICS_SUMMARY_SUM)
+        .filter(StatisticsShortTerm.start >= bindparam("start_time"))
+        .filter(StatisticsShortTerm.start < bindparam("end_time"))
+        .subquery()
+    )
+    query = (
+        session.query(subquery)
+        .filter(subquery.c.rownum == 1)
+        .order_by(subquery.c.metadata_id)
+    )
+    stats = execute(query.params(start_time=start_time, end_time=end_time))
```

Comment:
Should we put this behind a lambda_stmt since its going to run frequently (for a followup PR)?